azure-kusto-data/tests/test_kusto_client.py

```diff
@@ -343,3 +345,26 @@ def test_admin_then_query(self, mock_post):
         self.assertEqual(response[1].table_kind, WellKnownDataSet.QueryProperties)
         self.assertEqual(response[2].table_kind, WellKnownDataSet.QueryCompletionInformation)
         self.assertEqual(response[3].table_kind, WellKnownDataSet.TableOfContents)
+
+    @patch("urllib3.PoolManager.request", side_effect=mocked_poolmgr_request)
+    def test_dynamic(self, mock_post):
+        """Tests dynamic responses."""
+        client = KustoClient("https://somecluster.kusto.windows.net")
+        query = """print dynamic(123), dynamic("123"), dynamic("test bad json"), dynamic(null), dynamic('{"rowId":2,"arr":[0,2]}'), dynamic({"rowId":2,"arr":[0,2]})"""
+        row = client.execute_query("PythonTest", query).primary_results[0].rows[0]
+        self.assertIsInstance(row[0], int)
+        self.assertEqual(row[0], 123)
+
+        self.assertIsInstance(row[1], text_type)
+        self.assertEqual(row[1], "123")
+
+        self.assertIsInstance(row[2], text_type)
+        self.assertEqual(row[2], "test bad json")
+
+        self.assertEqual(row[3], None)
+
+        self.assertIsInstance(row[4], text_type)
+        self.assertEqual(row[4], '{"rowId":2,"arr":[0,2]}')
+
+        self.assertIsInstance(row[5], dict)
+        self.assertEqual(row[5], {"rowId": 2, "arr": [0, 2]})
```

Comment:
does this work? assert dict's doesn't usually work very well, I would replace it with asserting actual values inside the dict..